### DISCUSSION TITLE
High Re (unsteady effects) flow convergence with incomp solver

### DISCUSSION THREAD

**Discussion by NAnand-TUD** [su2code/SU2#1929] [General]
Hello Community,

I am trying to simulate a fin array using the INC_RANS solver with SST @ RE2000. However, the residual oscillates and does not converge sufficiently due to unsteady effects. Do you have recommendations to achieve convergence in such a flow?

Thanks
Nitish

PS: @bigfooted you probably have a handy suggestion.

**Comment by bigfooted** [comment]
Can you share the case on github?

**Comment by pcarruscag** [comment]
This is probably the issue we talked about @suargi @bigfooted, @suargi any updates?

**Reply by suargi** [reply]
My scenario is laminar instead of turbulent, I do not use periodic boundaries neither wall model (not really sure if you are using WM).

My residuals to not really oscillate. My issue seems to be related to very small dual time step, dt, values. When using rather small dt values some oscillations appear in the drag coefficient but not in the lift. As for my latest results, it seems that these oscillations belong to the transitory, as they have nearly disappeared after a long time. Apparently smaller dt has produces a considerably longer transitory. The situation gets worse when using a finer mesh. There, the drag coefficient could be even classified as erratic, while the lift coefficient is correct once again. I do not know what will happen after a long time running (it is in progress).

Regarding the residuals for each pseudo-time step, not matter the dt value:
 - Velocity: it converges 3 orders of magnitude.
 - Pressure: I am lucky if it converges one order of magnitude.
 The initial residuals values progressively get smaller as we progress further in time, but the behavior is the same all along the simulation.

I have tried different strategies to prevent the oscillations from appearing. Two solutions are (further investigation is necessary):
 - Set the BETA_FACTOR parameter in the config file to a very large value. This helped for 'very fine' meshes, but not for 'not that fine' meshes.
 - Set the BetaInc2 to a very larger value only in [CIncEulerSolver:SetPreconditioner](https://github.com/su2code/SU2/blob/b80e4ce0e4fd05afa136af0990a17dd619cdce42/SU2_CFD/src/solvers/CIncEulerSolver.cpp#L1937). This helped for 'not that fine' meshes but not for the 'very fine' ones.
 I do not know how to classify a mesh as 'very fine' or 'not that fine'. I arrived to the previous findings by just trying both approaches in different refined meshes.

**Reply by NAnand-TUD** [reply]
Thanks, @suargi for this detailed explanation. Can you please give me an indication of the values that you used? At least order of magnitude.

**Reply by suargi** [reply]
I have tried with $10^3, 10^4$ and $10^5$ for both approaches. The first two didn't work, the solution diverged. This last one worked.

**Comment by NAnand-TUD** [comment]
You can find the test case here.

https://github.com/NAnand-TUD/inline_fin_su2

Thanks,
Nitish
